src/main/java/fi/aalto/cs/intellij/activities/RequiredPluginsCheckerActivity.java

```diff
@@ -0,0 +1,213 @@
+   */
+  private void filterMissingOrDisabledPluginNames() {
+    missingOrDisabledPluginNames = requiredPluginNames
+        .entrySet()
+        .stream()
+        .filter(entry -> isPluginMissingOrDisabled(entry.getValue()))
+        .collect(toMap(Entry::getKey, Entry::getValue));
+  }
+
+  /**
+   * Predicate for checking the plugin status.
+   */
+  private static boolean isPluginMissingOrDisabled(String id) {
+    PluginId pluginId = PluginId.getId(id);
+    return !PluginManager.isPluginInstalled(pluginId) || PluginManager
+        .isDisabled(pluginId.getIdString());
+  }
+
+  /**
+   * If there are any plugins missing, creates a list of the plugin descriptors for them based on the
+   * publicly available ones.
+   */
+  private void createListOfMissingOrDisabledPluginDescriptors() {
+    if (missingOrDisabledPluginNames.size() > 0) {
+      getAvailablePluginsFromMainRepo();
+      missingOrDisabledIdeaPluginDescriptors = new ArrayList<>();
+      missingOrDisabledPluginNames.forEach((name, id) -> {
+        availableIdeaPluginDescriptors
+            .stream()
+            .filter(
+                availableDescriptor -> availableDescriptor.getPluginId().equals(PluginId.getId(id))
+            )
+            .findFirst()
+            .ifPresent(missingOrDisabledIdeaPluginDescriptors::add);
+      });
+    }
+  }
+
+  /**
+   * Get the list of all the available in the main JetBrains plugin repository.
+   *
+   * <p>Note! The descriptors for the not installed plugins do not exist in IJ until now.
+   */
+  private void getAvailablePluginsFromMainRepo() {
+    try {
+      availableIdeaPluginDescriptors = RepositoryHelper.loadPlugins(new BgProgressIndicator());
+    } catch (IOException ex) {
+      logger.error("Could not retrieve plugins data from main repository.", ex);
+    }
+  }
+
+  /**
+   * Sorts required plugins into missing and disabled and shows respective notifications.
+   */
+  private void checkPluginsStatusAndNotify() {
+    List<IdeaPluginDescriptor> missingPluginDescriptors = new ArrayList<>();
+    List<IdeaPluginDescriptor> disabledPluginDescriptors = new ArrayList<>();
+
+    for (IdeaPluginDescriptor descriptor : missingOrDisabledIdeaPluginDescriptors) {
+      if (!PluginManager.isPluginInstalled(descriptor.getPluginId())) {
+        missingPluginDescriptors.add(descriptor);
+      } else if (PluginManager.isDisabled(descriptor.getPluginId().getIdString())) {
+        disabledPluginDescriptors.add(descriptor);
+      }
+    }
+
+    if (missingPluginDescriptors.size() > 0) {
+      notifyAndSuggestPluginsInstallation(missingPluginDescriptors);
+    } else if (disabledPluginDescriptors.size() > 0) {
+      notifyAndSuggestPluginsEnabling(disabledPluginDescriptors);
+    }
+  }
+
+  /**
+   * Notify with an option to enable all the required plugins.
+   */
+  private void notifyAndSuggestPluginsEnabling(List<IdeaPluginDescriptor> descriptors) {
+    Notification notification = new Notification(
```

Comment:
I'd consider extracting this to a separate method, e.g. "notifyAboutDisabledPlugins".